extensions/amp-story/1.0/progress-bar.js

```diff
@@ -75,45 +79,72 @@ export class ProgressBar {
   }
 
   /**
-   * @param {!Array<string>} segmentIds The id of each segment in the story.
+   * Builds the progress bar.
+   *
    * @return {!Element}
    */
-  build(segmentIds) {
+  build() {
     if (this.isBuilt_) {
       return this.getRoot();
     }
 
-    const segmentCount = segmentIds.length;
-    devAssert(segmentCount > 0);
-
     this.isBuilt_ = true;
-    this.segmentCount_ = segmentCount;
-
-    segmentIds.forEach((id, i) => (this.segmentIdMap_[id] = i));
 
     this.root_ = this.win_.document.createElement('ol');
     this.root_.classList.add('i-amphtml-story-progress-bar');
 
-    for (let i = 0; i < this.segmentCount_; i++) {
-      const segmentProgressBar = this.win_.document.createElement('li');
-      segmentProgressBar.classList.add('i-amphtml-story-page-progress-bar');
-      const segmentProgressValue = this.win_.document.createElement('div');
-      segmentProgressValue.classList.add('i-amphtml-story-page-progress-value');
-      segmentProgressBar.appendChild(segmentProgressValue);
-      this.root_.appendChild(segmentProgressBar);
-    }
+    this.storeService_.subscribe(
+      StateProperty.STORY_PAGE_IDS,
+      pageIds => {
+        pageIds.forEach(id => {
+          if (!(id in this.segmentIdMap_)) {
```

Comment:
Nit: `if (!this.segmentIdMap_[id])`

Reply:
I tried this before but it breaks when `map[id] = 0`. Because `!0 === true`.  Because JS...

Good thing is I found several usages of `!(id in map)` in the amp codebase so it should be fine.